ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/prefix_lists.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from .utils import UtilsMixin
+
+
+class PrefixListsMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def prefix_lists(self) -> dict | None:
+        """
+        Return structured config for prefix_lists
+        """
+        if self._underlay_router is not True:
```

Comment:
so what would be the way to simplify using these? not setting cached_property for these variables separately and loading the `self._underlay` and `self._overlay`?

I would rather we do it right in this PR and move forward to avoid too much rework later on :)

Reply:
You could map out _underlay_bgp, _underlay_isis, _underlay_ospf, _overlay_evpn one by one, I did that for mine. Or you could just use cached_property for the dict and check that. Just remember to use `self._underlay["ospf"]`. Not as clean IMO.